sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -897,4 +913,162 @@ public ServiceBusReceiverClient buildClient() {
+         *
+         * @param receiveMode Mode for receiving messages.
+         *
+         * @return The modified {@link ServiceBusDeadLetterReceiverClientBuilder} object.
+         */
+        public ServiceBusDeadLetterReceiverClientBuilder receiveMode(ReceiveMode receiveMode) {
+            this.receiveMode = receiveMode;
+            return this;
+        }
+
+        /**
+         * Sets the name of the subscription in the topic to listen to. <b>{@link #topicName(String)} must also be set.
+         * </b>
+         *
+         * @param subscriptionName Name of the subscription.
+         *
+         * @return The modified {@link ServiceBusDeadLetterReceiverClientBuilder} object.
+         * @see #topicName A topic name should be set as well.
+         */
+        public ServiceBusDeadLetterReceiverClientBuilder subscriptionName(String subscriptionName) {
+            this.subscriptionName = subscriptionName;
+            return this;
+        }
+
+        /**
+         * Sets the name of the topic. <b>{@link #subscriptionName(String)} must also be set.</b>
+         *
+         * @param topicName Name of the topic.
+         *
+         * @return The modified {@link ServiceBusDeadLetterReceiverClientBuilder} object.
+         * @see #subscriptionName A subscription name should be set as well.
+         */
+        public ServiceBusDeadLetterReceiverClientBuilder topicName(String topicName) {
+            this.topicName = topicName;
+            return this;
+        }
+
+        /**
+         * Creates an <b>asynchronous</b> Service Bus receiver responsible for reading {@link ServiceBusMessage
+         * messages} from a specific queue or topic.
+         *
+         * @return An new {@link ServiceBusReceiverAsyncClient} that receives messages from a queue or topic.
+         * @throws IllegalStateException if {@link #queueName(String) queueName} or {@link #topicName(String)
+         *     topicName} are not set or, both of these fields are set. It is also thrown if the Service Bus {@link
+         *     #connectionString(String) connectionString} contains an {@code EntityPath} that does not match one set in
+         *     {@link #queueName(String) queueName} or {@link #topicName(String) topicName}. Lastly, if a {@link
+         *     #topicName(String) topicName} is set, but {@link #subscriptionName(String) subscriptionName} is not.
+         * @throws IllegalArgumentException Queue or topic name are not set via {@link #queueName(String)
+         *     queueName()} or {@link #topicName(String) topicName()}, respectively.
+         */
+        public ServiceBusReceiverAsyncClient buildAsyncClient() {
+            final MessagingEntityType entityType = validateEntityPaths(logger, connectionStringEntityName, topicName,
+                queueName);
+            final String entityPath = getDeadLetterEntityPath(logger, entityType, queueName, topicName,
+                subscriptionName);
+
+            if (prefetchCount < 1) {
```

Comment:
The amount of duplicated logic we're amassing is concerning. This is going to be painful from a maintenance standpoint.